UnityProject/Assets/Scripts/Shuttles/EscapeShuttleConsole.cs

```diff
@@ -72,17 +83,70 @@ private void FinishHack(HandApply interaction)
+		}
+
+		private void RegisterEarlyShuttleLaunch(IDCard card, PlayerScript performer)
+		{
+			if (GameManager.Instance.ShuttleSent)
+			{
+				Chat.AddExamineMsg(performer.gameObject, "The shuttle is already moving!");
+				return;
+			}
+			if (registeredIDs.Contains(card))
+			{
+				Chat.AddExamineMsg(performer.gameObject, "You've already done this!");
+				return;
+			}
+			registeredIDs.Add(card);
+
+			if (GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta && registeredIDs.Count >= 2)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			if (registeredIDs.Count >= 4)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			AnnounceRemainingSwipesRequired();
+		}
+
+		private void AnnounceRemainingSwipesRequired()
+		{
+			var swpiesRequired = GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta ? 2 : 4;
+			var remainingSwipes = swpiesRequired - registeredIDs.Count;
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch has been request! need {remainingSwipes} more votes.</b></size></color>\n\n",
+				MatrixManager.MainStationMatrix);
+
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch has been request! need {remainingSwipes} more votes.</b></size></color>\n\n",
+				GameManager.Instance.PrimaryEscapeShuttle.MatrixInfo);
+			_ = SoundManager.PlayNetworked(CommonSounds.Instance.Notice1);
+		}
 
-			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered!</b></size></color>\n\n",
+		private void DepartShuttle()
+		{
+			if (GameManager.Instance.ShuttleSent) return;
+			var departTime = beenEmagged ? 5 : 10;
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered! Launching in {departTime} seconds..</b></size></color>\n\n",
 				MatrixManager.MainStationMatrix);
 
-			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered!</b></size></color>\n\n",
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch Triggered! Launching in {departTime} seconds..</b></size></color>\n\n",
 				GameManager.Instance.PrimaryEscapeShuttle.MatrixInfo);
 
 			_ = SoundManager.PlayNetworked(CommonSounds.Instance.Notice1);
+			GameManager.Instance.ForceSendEscapeShuttleFromStation(departTime);
+		}
 
-			GameManager.Instance.ForceSendEscapeShuttleFromStation(10);
+		public RightClickableResult GenerateRightClickOptions()
+		{
+			if (AdminCommandsManager.IsAdmin(PlayerManager.LocalPlayerScript.netIdentity.connectionToServer,
+				    out var _) == false) return null;
+			var result = RightClickableResult.Create().AddElement("Launch Early", DepartShuttle, Color.red);
```

Comment:
Does this work in multiplayer? I think right click options are executed on the client.

Reply:
How does it not matter if it doesn't work in the actual game?

Reply:
> How does it not matter if it doesn't work in the actual game?

Because its mainly meant for testing, i don't think admins should be able to push the shuttle early from a console that they probably cant even access from their ghost view. It should be from their tools or a console command.

Reply:
Huh, I guess in that case it's fine.